src/main/java/ch/njol/skript/expressions/ExprNamed.java

```diff
@@ -35,51 +35,85 @@
 import ch.njol.skript.expressions.base.PropertyExpression;
 import ch.njol.skript.lang.Expression;
 import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.Literal;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.registrations.Classes;
 import ch.njol.skript.util.Getter;
 import ch.njol.util.Kleenean;
 
-/**
- * @author Peter Güttinger
- */
 @Name("Named Item/Inventory")
-@Description("Directly names an item/inventory, useful for defining a named item/inventory in a script. " +
-		"If you want to (re)name existing items/inventories you can either use this expression or use <code>set <a href='#ExprName'>name of &lt;item/inventory&gt;</a> to &lt;text&gt;</code>.")
-@Examples({"give a diamond sword of sharpness 100 named \"&lt;gold&gt;Excalibur\" to the player",
-		"set tool of player to the player's tool named \"&lt;gold&gt;Wand\"",
-		"set the name of the player's tool to \"&lt;gold&gt;Wand\"",
-		"open hopper inventory named \"Magic Hopper\" to player"})
+@Description({
+	"Directly names an item/inventory, useful for defining a named item/inventory in a script.",
+	"If you want to (re)name existing items/inventories you can either use this expression " +
+			"or use <code>set <a href='#ExprName'>name of &lt;item/inventory&gt;</a> to &lt;text&gt;</code>."
+})
+@Examples({
+	"give a diamond sword of sharpness 100 named \"&lt;gold&gt;Excalibur\" to the player",
+	"set tool of player to the player's tool named \"&lt;gold&gt;Wand\"",
+	"set the name of the player's tool to \"&lt;gold&gt;Wand\"",
+	"open hopper inventory named \"Magic Hopper\" to player"
+})
 @Since("2.0, 2.2-dev34 (inventories)")
 public class ExprNamed extends PropertyExpression<Object, Object> {
+
 	static {
-		Skript.registerExpression(ExprNamed.class, Object.class, ExpressionType.PROPERTY,
+		Skript.registerExpression(ExprNamed.class, Object.class, ExpressionType.COMBINED,
 				"%itemtype/inventorytype% (named|with name[s]) %string%");
 	}
-	
-	@SuppressWarnings("null")
+
 	private Expression<String> name;
-	
-	@SuppressWarnings({"unchecked", "null"})
+
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		setExpr(exprs[0]);
 		name = (Expression<String>) exprs[1];
+		if (getExpr() instanceof Literal) {
+			Literal<?> literal = (Literal<?>) getExpr();
+			Object object = literal.getSingle();
+			if (object instanceof InventoryType && !((InventoryType) object).isCreatable()) {
+				Skript.error("You can't create a '" + literal.toString() + "' inventory. It's not creatable!");
```

Comment:
We could also check the offending bookshelf/composter inventories here for 1.19.4 :)

Reply:
Can't because bookshelf/composter inventories enum don't exist on older versions like 1.13.